state-chain/primitives/src/lib.rs

```diff
@@ -13,10 +13,58 @@ use serde::{Deserialize, Serialize};
 use sp_core::{ConstU32, U256};
 use sp_std::{
 	cmp::{Ord, PartialOrd},
+	fmt,
+	ops::{Deref, DerefMut},
 	vec::Vec,
 };
 pub mod chains;
 
+macro_rules! define_wrapper_type {
+	($name: ident, $inner: ty) => {
+		#[derive(
+			Clone,
+			Copy,
+			RuntimeDebug,
+			PartialEq,
+			Eq,
+			Encode,
+			Decode,
+			TypeInfo,
+			MaxEncodedLen,
+			Default,
+			serde::Serialize,
+			serde::Deserialize,
```

Comment:
FWIW this might cause issues in pallets/crates that don't include serde. Maybe we can let users add their own?

Not sure this works?

```rust
#[derive(X)]
define_wrapper_type!(SwapId, u64);
```

Might have to be:

```rust
define_wrapper_type!(SwapId, u64, extra_derives: X, Y, Z);
```